File: problems/intermediate/power_ranger/power_ranger.java
package problems.intermediate.power_ranger;

public class power_ranger {
    public static int getPowerRange(int n, int a, int b) {
        int count = 0;
        int x = 1;

        while (true) {
            double power = Math.pow(x, n);
            if (power > b) {
                break;
            }
            if (power >= a) {
                count++;
            }
            x++;
        }
        return count;
    }

    public static void main(String[] args) {
        System.out.println(getPowerRange(2, 49, 65));
        System.out.println(getPowerRange(3, 1, 27));
        System.out.println(getPowerRange(10, 1, 5)); 
        System.out.println(getPowerRange(5, 31, 33));
        System.out.println(getPowerRange(4, 250, 1300)); 
    }
}

File: problems/intermediate/power_ranger/power_ranger.py
def getPowerRange(n, a, b):
    count = 0
    x = 1

    while True:
        power = x ** n
        if power > b:
            break
        if power >= a:
            count += 1
        x += 1

    return count


print(getPowerRange(2, 49, 65))  
print(getPowerRange(3, 1, 27))     
print(getPowerRange(10, 1, 5))     
print(getPowerRange(5, 31, 33))    
print(getPowerRange(4, 250, 1300)) 
